osdep/win32/pthread.c

```diff
@@ -96,11 +95,22 @@ int pthread_cond_timedwait(pthread_cond_t *restrict cond,
                            pthread_mutex_t *restrict mutex,
                            const struct timespec *restrict abstime)
 {
-    // mp time is not converted to realtime if internal pthread impl is used
-    int64_t now = mp_time_ns();
-    int64_t time_ns = abstime->tv_sec * UINT64_C(1000000000) + abstime->tv_nsec;
-    int64_t timeout_ms = (time_ns - now) / INT64_C(1000000);
-    return cond_wait(cond, mutex, MPCLAMP(timeout_ms, 0, INFINITE));
+    // mpv uses mingw's gettimeofday() as time source too.
+    struct timeval tv;
+    gettimeofday(&tv, NULL);
```

Comment:
does macOS actually not support CLOCK_REALTIME and the associated functions or is that only for totally outdated versions?

Reply:
They added it in 10.12. The README claims that we support macOS 10.8 (no idea if that's actually true), so we ended up keeping the legacy stuff in there.

Reply:
imo we can just kill that support. the oldest OS i tested my changes on actively was 10.12 in a VM, otherwise it was 10.14.

though no idea how good CLOCK_REALTIME works.